chemicalx/utils.py

```diff
@@ -0,0 +1,58 @@
+"""Utilities for chemicalx."""
```

Comment:
please provide unit tests for these helper functions

Reply:
Provided tests through the softmax segment - that reuses both.